app/routes.py

```diff
@@ -1 +1,280 @@
+def uncheck_task(task_id):
+    task = validate_item(Task,task_id)
+    task.completed_at=None
+
+    db.session.commit()
+
+    return make_response({"task": to_dict(task)})
+
+#---------------------------------------------------------------#
+#-------------------------GOAL ROUTES---------------------------#
+#---------------------------------------------------------------#
+
+#-------------GET ALL GOALS------------#
+@goal_bp.route("", methods =["GET"])
+def get_goals():
+    goals = query_filter(Goal)
+    goals_response = [to_dict(goal) for goal in goals]
+
+    return make_response(jsonify(goals_response), 200)
+
+#-------------MAKE NEW GOAL-------------#
+@goal_bp.route("", methods =["POST"])
+def make_goal():
+    request_body = request.get_json()
+    new_goal = validate_new_item(Goal,request_body)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return make_response({"goal":to_dict(new_goal)}, 201)
+
+#-------------GET SINGLE GOAL------------#
+@goal_bp.route("/<goal_id>", methods=["GET"])
+def one_goal(goal_id):
+    goal = validate_item(Goal,goal_id)
+
+    return make_response(dict(goal=to_dict(goal)), 200)
+
+#-------------EDIT SINGLE GOAL-------------#
+@goal_bp.route("/<goal_id>", methods=["PUT"])
+def edit_goal(goal_id):
+    goal = validate_item(Goal,goal_id)
+    request_body = request.get_json()
+
+    if "title" in request_body:
+        goal.title=request_body["title"]
+    if "description" in request_body:
+        goal.description=request_body["description"]
+
+    db.session.commit()
+
+    return(make_response({"goal":to_dict(goal)}, 200))
+
+#-------------DELETE SINGLE GOAL-------------#
+@goal_bp.route("/<goal_id>", methods=["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_item(Goal,goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return make_response({"details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'},200)
+
+#---------------------------------------------------------------#
+#-----------------------NESTED ROUTES---------------------------#
+#---------------------------------------------------------------#
+
+#-------------GET ALL GOAL'S TASKs-------------#
+
+@goal_bp.route("/<goal_id>/tasks", methods=["GET"])
+def get_tasks_for_goal(goal_id):
+    goal = validate_item(Goal,goal_id)
+    goal_response = to_dict(goal,list_tasks=True)
+
+    return make_response(jsonify(goal_response), 200)
+
+#---------------ADD TASKS TO GOAL--------------#
+@goal_bp.route("/<goal_id>/tasks", methods=["POST"])
+def add_tasks(goal_id):
+    goal = validate_item(Goal,goal_id)
+    request_body = request.get_json()
+    
+    if "task_ids" in request_body:
+        task_list = [task_to_goal(id,goal) for id in request_body["task_ids"]]
+    else:
+        abort(make_response({"details": "Invalid data"},400))
+
+    db.session.commit()
+
+    return make_response({"id": goal.goal_id, "task_ids": task_list},200)
+
```

Comment:
This isn't too big of a deal for a project of this size, but it's a generally good practice to separate routes into their own files. Then the helper functions that are common to all can be extracted to their own file and imported in each route file.